Duplicati/Library/Utility/Utility.cs

```diff
@@ -1013,11 +1014,39 @@ public static string ExpandEnvironmentVariablesRegexp(string str, Func<string, s
                 //IsClientLinux ? ENVIRONMENT_VARIABLE_MATCHER_LINUX : ENVIRONMENT_VARIABLE_MATCHER_WINDOWS
 
                 ENVIRONMENT_VARIABLE_MATCHER_WINDOWS
-                    .Replace(str.Replace("~", Regex.Escape(HOME_PATH)), (m) => 
+                    .Replace(Regex.Escape(ExpandTilde(str)), (m) => 
                         Regex.Escape(lookup(m.Groups["name"].Value)));
         }
 
         /// <summary>
+        /// On Unix-like systems, expand the '~' character to the user's home directory.
+        /// </summary>
+        /// <param name="path">The path to expand.</param>
+        /// <returns>An expanded version of <paramref name="path"/>, with '~' replaced by the user's home directory.</returns>
+        public static string ExpandTilde(string path)
+        {
+            // We only replace '~' on Unix-like systems to avoid potential issues
+            // with paths like c:\Progra~1\ on Windows.
+            return IsClientLinux ? path.Replace("~", HOME_PATH) : path;

```

Comment:
I think this will break some Windows things, where `~` is expanded to the user folder?

Reply:
Sorry, I'm not as familiar with Windows.  In what contexts does `~` represent the user folder on Windows?

Reply:
In fact, my implementation here isn't quite correct on Linux either.  In most shells, a single `~` or a path beginning with `~/` is expanded to the current user's home directory.  A tilde followed by a username (e.g., `~user`) is expanded to the specified user's home directory.  Moreover, `~` is a valid character for file and directory names on many filesystems, so `~` is a valid filename and `~user` is a valid directory name.

As such, I think we can reasonably handle the following cases:
1. If `~` is not the first character of the string, do not perform any expansion.
2. If the string begins with `~/`, expand it to the user's home directory.  I'm not sure if this is typical on Windows, but we can certainly adopt this as a convention. 

Any other case seems ambiguous to me.  It's difficult to tell if `~user` refers to `user`'s home directory, or a file or directory named `~user`.  For performance reasons, I don't think we would want to do anything too fancy to resolve these.

What do you think is the best way to proceed?

Reply:
Good point.  It seems that tilde expansion is a bit more complicated and deserves a separate pull request.  Since I was able to resolve my problem with the redundant path separators (fixed my `HOME` environment variable), I'll create a new issue for the tilde expansion and close this pull request.